bundles/org.openhab.binding.synopanalyzer/src/main/java/org/openhab/binding/synopanalyzer/internal/SynopAnalyzerHandlerFactory.java

```diff
@@ -49,6 +76,42 @@ public boolean supportsThingType(ThingTypeUID thingTypeUID) {
     protected @Nullable ThingHandler createHandler(Thing thing) {
         ThingTypeUID thingTypeUID = thing.getThingTypeUID();
 
-        return thingTypeUID.equals(THING_SYNOP) ? new SynopAnalyzerHandler(thing) : null;
+        return thingTypeUID.equals(THING_SYNOP) ? new SynopAnalyzerHandler(thing, locationProvider, stationDB) : null;
+    }
+
+    @Override
+    protected void activate(ComponentContext componentContext) {
+        super.activate(componentContext);
+        InputStream is = Thread.currentThread().getContextClassLoader().getResourceAsStream("/db/stations.json");
+        try {
+            Reader reader = new InputStreamReader(is, "UTF-8");
```

Comment:
Put it in a try-with-resources (and you can remove the `close()`):
```suggestion
        try (InputStream is = Thread.currentThread().getContextClassLoader().getResourceAsStream("/db/stations.json"); 
            Reader reader = new InputStreamReader(is, "UTF-8");) {
```
Also run mvn spotless:apply just to make sure it formats correctly.